app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            "{"
+            + f"Name: {self.name}, "
+            + f"Health: {self.health}, "
+            + f"Hidden: {self.hidden}"
+            + "}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(victim: Herbivore) -> None:
+        if isinstance(victim, Herbivore) and not victim.hidden:
+            victim.health -= 50
+        if victim.health <= 0:
+            for animal in range(len(Animal.alive)):
+                if Animal.alive[animal] is victim:
+                    del Animal.alive[animal]
```

Comment:
Seriously, it is possible that another animal than victim can have hp <= 0 due to this method?

Reply:
It's not possible, but I need to remove the dead victim from the list, and for this I need to find it, so I'm looking for it by the index

Reply:
Just use `.remove()` method and victim as an argument

Reply:
Wow, that's the method I missed, thanks